pkg/internal/cluster/logs/logs.go

```diff
@@ -127,6 +113,46 @@ func Collect(nodes []nodes.Node, dir string) error {
 	return errors.NewAggregate(errs)
 }
 
+// dumpDir dumps the dir nodeDir on the node to the dir hostDir on the host
+func dumpDir(node nodes.Node, nodeDir, hostDir string) (err error) {
+	// make tempdir to rsync nodeDir into (rsync handles taking a snapshot better)
+	tmp, err := mktemp(node)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if rerr := node.Command("rm", "-rf", tmp).Run(); rerr != nil && err == nil {
+			err = rerr
```

Comment:
if there is an error here, it will be passed to the calling function?
 reading https://golang.org/ref/spec#Defer_statements, `If the deferred function has any return values, they are discarded when the function completes`

Reply:
Yes, but only if we didn't already have an error